metaflow/datastore/datastore.py

```diff
@@ -441,9 +441,10 @@ def _save_object(self, transformable_obj, var, force_v4=False):
                                         % var)
                 transformable_obj.transform(lambda x: pickle.dumps(x, protocol=4))
         sha = sha1(transformable_obj.current()).hexdigest()
-        sz = len(transformable_obj.current())
-        self.save_data(sha, transformable_obj)
-        return sha, sz, blobtype
+        pickle_sz = len(transformable_obj.current())
+        path = self.save_data(sha, transformable_obj)
+        unpickle_sz = sys.getsizeof(open(path, 'rb'))
```

Comment:
Path is not necessarily local (could be on S3) so this won't work. Even if local, this may be a GZip path.